test/functional/api/vim_spec.lua

```diff
@@ -81,6 +81,26 @@ describe('api', function()
     end)
   end)
 
+  describe('nvim_execute_lua', function()
```

Comment:
By that argument we could test any api function through call_atomic. Why do we expect execute_lua to have any special "interactions"? 

If lua is used for batching and coordination, then how is call_atomic needed?